BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsClient.kt

```diff
@@ -19,13 +19,15 @@ internal class AnalyticsClient @VisibleForTesting constructor(
     private val httpClient: BraintreeHttpClient,
     private val analyticsDatabase: AnalyticsDatabase,
     private val workManager: WorkManager,
-    private val deviceInspector: DeviceInspector
+    private val deviceInspector: DeviceInspector,
+    private val isVenmoInstalled: Boolean
 ) {
     constructor(context: Context) : this(
         BraintreeHttpClient(),
         getInstance(context.applicationContext),
         WorkManager.getInstance(context.applicationContext),
-        DeviceInspector()
+        DeviceInspector(),
+        DeviceInspector().isVenmoInstalled(context.applicationContext)
```

Comment:
This creates a new instance of device inspector, do we really want it? Can we reuse the device inspector we already have?

Reply:
I couldn't figure out a way to do iOS equivalent of static function for device install check that I implemented (https://github.com/braintree/braintree_ios/pull/1289) 
I will try `companion object`

Reply:
We could add the `deviceInspector.isVenmoInstalled()` check [here](https://github.com/braintree/braintree_android/blob/426544a92c4729fe97d299f9166a14602b0eb112/BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.kt#L222). Then we wouldn't need to create an `updatedEvent` internally.

Reply:
Nice!

Reply:
one issue I ran into is getting `venmoInstalled` in `writeAnalytics` function in AnalyticsClient without `isVenmoInstalled` 
property on AnalyticsClient.

Reply:
Yes, without i`sVenmoInstalled` property on `AnalyticsClient`, I ran into issue of accessing `venmoInstalled` value in instantiating `AnalyticEvent` in `writeAnalytics` function in `AnalyticsClient`.